### PR TITLE
feat(optimus)!: [RND-103442] Added new variants for Tags

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

RND-103442

Added missing variants for the Tag component.

**BREAKING CHANGE:** Bold/Subtle variants are replaced with OptimusTag (system, prev. bold) and OptimusCategoricalTag (data, prev. subtle). The signature of OptumusTag was changed.

Screenshots are below, hidden under spoilers. 

<details>
<summary>Light Mode</summary>

- Was
Screenshot 2022-05-11 at 12 05 20

- Is
Screenshot 2022-05-11 at 12 00 58
</details>

<details>
<summary>Dark Mode</summary>

- Was
Screenshot 2022-05-11 at 12 15 11

- Is
Screenshot 2022-05-12 at 13 33 59
</details>

#### Testing steps

No.

#### Follow-up issues

No.

#### Check during review

- Verify against YouTrack issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

**Comment by ookami-kb** [comment]
@witwash the build is failing.

File: optimus/lib/src/tag.dart [hunk 1]
```diff
@@ -151,89 +143,166 @@ class _TagState extends State<_Tag> with ThemeGetter {
...
+  denim,
+  lavender,
+  lime,
+  mustard,
+  ruby,
+  tangerine
+}
+
+/// Color options are designed so they won't carry any semantic meaning. Could be used in any cause when
+/// displaying categorical data.
```
Comment (ookami-kb) [inline optimus/lib/src/tag.dart:218]: Add extra line after this line.

File: optimus/lib/src/tag.dart [hunk 2]
```diff
@@ -151,89 +143,166 @@ class _TagState extends State<_Tag> with ThemeGetter {
...
+/// [OptimusCategoricalColorOption.denim] - Denim Blue
+/// [OptimusCategoricalColorOption.lavender] - Lavender Purple
+/// [OptimusCategoricalColorOption.lime] - Lime Green
+/// [OptimusCategoricalColorOption.mustard] - Mustard Yellow
+/// [OptimusCategoricalColorOption.ruby] - Ruby Red
+/// [OptimusCategoricalColorOption.tangerine] - Tangerine Orange
+///
```
Comment (ookami-kb) [inline optimus/lib/src/tag.dart:225]: Remove this extra line.

File: optimus/lib/src/tag.dart [hunk 3]
```diff
@@ -151,89 +143,166 @@ class _TagState extends State<_Tag> with ThemeGetter {
...
+          color: _textColor(theme),
+          fontWeight: FontWeight.w600,
+          fontSize: 12,
+        ),
+        overflow: TextOverflow.ellipsis,
+      ),
+    );
+  }
+
+  Color get _borderColor {
```
Comment (ookami-kb) [inline optimus/lib/src/tag.dart:259]: Let's rather convert them to extensions on `OptimusCategoricalColorOption`.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by ookami-kb** [comment]
@witwash please also add the screenshot of the storybook to the PR description – it's easier to get the context of the issue.

**Comment by witwash** [comment]
Done.

p.s. Is it me or does dark mode look off? It's less readable

**Comment by ookami-kb** [comment]
@witwash yeah, I think dark mode looks weird. Can you please confirm it with UX guys?

**Comment by witwash** [comment]
Sure thing. Will keep you updated.

**Comment by witwash** [comment]
<details>
<summary>New Dark Colors </summary>
Screenshot 2022-05-12 at 13 33 59

</details>

**Comment by ookami-kb** [comment]
@witwash please update the PR description with new screenshots

**Pull request merged**